### PR TITLE
HELM-277: flow dashboard update for Grafana 7

### PR DESCRIPTION
**Pull request by mmahacek**
# Pull Request Check Sheet

* [X] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [X] Have you made a comment in that issue which points back to this PR?
* [X] Have you updated the JIRA link at the bottom of this comment to link to your issue?

# External References

* JIRA (Issue Tracker): https://issues.opennms.org/projects/HELM/issues/HELM-277
* Continuous Integration: [CircleCI](https://circleci.com/gh/OpenNMS/opennms-helm)

### CODE REVIEW

**Review by swachter — COMMENTED** [review]
What to review here? Files changed: 0

**Comment by kramrm** [comment]
Nothing. I forgot to retract this PR as I need to redo my changes to incorporate your dscp updates to the dashboard.

**Comment by mmahacek** [comment]
I need to re-do my changes since HELM-258 has been merged.

**Review by swachter — COMMENTED** [review]
I tried the new dashboard. It seems to work as expected. However, I wonder if the `negativeEgress` transformation could be used to achieve a similar effect. When trying it, it seemed not to work as expected, though. Unfortunately, I am not familiar with that transform and the expected visual result.

Can you check the `negativeEgress` transform?

Screenshot from 2021-05-27 10-07-11

**Review by indigo423 — COMMENTED** [review]
LGTM! In my modifications I would use the "negativeEgress" as @swachter mentioned. I used mainly the "negativeEgress" instead the Grafana built-in transform method in my modifications as well.

**Comment by mmahacek** [comment]
I was having issues getting outbound flows logged in my system for testing this.  I'll try again to see if I can get some generated to test `negativeEgress` as an alternative.

File: src/dashboards/flow_deep_dive.json [hunk 1]
```diff
@@ -323,11 +406,19 @@
         }
       ],
       "title": "Data Usage by Application",
-      "transform": "table",
+      "transformations": [
```
Comment (swachter) [inline src/dashboards/flow_deep_dive.json:409]: The `Merge` transformations at the summary tables must be removed. There seems a bug that prevents the value in the DSCP column to display properly.

Was there a specific reason for adding the `Merge` transformations?
Screenshot from 2021-05-27 21-16-16
Screenshot from 2021-05-27 21-16-25

**Review by swachter — CHANGES_REQUESTED** [review]
The `Merge` transformation must be removed.

Would it be feasible to have a minimal change of the `flow_deep_dive.json` file? Maybe you can just add

```
      "seriesOverrides": [
        {
          "alias": "/In/",
          "stack": "A"
        },
        {
          "alias": "/Out/",
          "stack": "B",
          "transform": "negative-Y"
        }
      ],
```

at the proper locations. I assume that the `$$hashKey` entries can be omitted.

File: src/dashboards/flow_deep_dive.json (hunk 1 above)
Comment (mmahacek) [reply src/dashboards/flow_deep_dive.json:409]: Not a specific reason for Merge. It was added when the dashboards updated to a more recent Grafana.  As Helm 7 now relies on Grafana 7.5, I left in dashboard schema changes generated by Grafana 7.5.

**Comment by mmahacek** [comment]
With series override to transform: negative-Y:
image

With negativeEgress:
image

**Review by swachter — COMMENTED** [review]
I looked at the flow-deep-dive dashboard. It seems to work as expected.

Unfortunately, the diff is so large, that it is difficult to review. I would prefer, if this PR would contain just minimal changes to the previous dashboard source (adding the the `negativeEgress` transformation and the stacking). If that is too much of an effort then I can also approve this PR, though.

**Comment by mmahacek** [comment]
@swachter Part of the reason for so many changes is that some of the panels used (built in Table panel) are considered legacy Grafana panels that are being phased out and replaced with a new built in table plugin, and this update includes that upgrade too.

**Review by swachter — APPROVED** [review]
Ok, then I approve.

**Pull request merged**